### DISCUSSION TITLE
How to install only devDependencies?

### DISCUSSION THREAD

**Discussion by romap0** [npm/feedback#416] [CLI] [answered]
- `npm ci --only=dev` installs 0 packages

- `npm ci --production=false` installs all dependencies

Is there any way to install only devDependencies?

Use case: Run eslint or unit tests in CI job

**Comment by ljharb** [comment]
How can you run unit tests without production dependencies installed? (also if you're using eslint-plugin-import, you'd need all your prod deps installed so the no-unresolved rule is satisfied)

**Reply by romap0** [reply]
For unit tests you don`t need any dependencies except the test runner itsel
no-unresolved rule will not work indeed

**Reply by ljharb** [reply]
Sure you do - you need any dependencies the "file under test" uses, which are all production dependencies.

**Comment by MylesBorins** [comment, marked as answer]
As mentioned by @ljharb I can't really see how a "devDep only" install would make sense in most environments, at least not enough to make sense as a platform supported feature. In the case of things like a linter in CI you could setup a CI job to only install those devDeps. If you have a large test runner or other devDeps it would seem like you wouldn't want to install those either.

If you would really like to pursue this proposal you should open an issue at https://github.com/npm/rfcs to get direct feedback from the CLI team
